File: Semester_04/Web/Exam/jsp-products/src/main/java/dto/DeveloperData.java
package dto;

public class DeveloperData {
  public int id;
  public String name;
  public int age;
  public String skills;

  public DeveloperData() {
  }

  public DeveloperData(int id, String name, int age, String skills) {
    this.id = id;
    this.name = name;
    this.age = age;
    this.skills = skills;
  }
}

File: Semester_04/Web/Exam/php-topics-notifications/dto/DeveloperData.php
<?php
class DeveloperData {
    public $id;
    public $name;
    public $age;
    public $skills;

    public function __construct($id = 0, $name = '', $age = 0, $skills = '') {
        $this->id = $id;
        $this->name = $name;
        $this->age = $age;
        $this->skills = $skills;
    }

    public function toArray() {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'age' => $this->age,
            'skills' => $this->skills
        ];
    }

    public static function fromArray($data) {
        return new DeveloperData(
            isset($data['id']) ? $data['id'] : 0,
            isset($data['name']) ? $data['name'] : '',
            isset($data['age']) ? $data['age'] : 0,
            isset($data['skills']) ? $data['skills'] : ''
        );
    }
}
?> 